client/platform/desktop/backend/native/common.ts

```diff
@@ -711,53 +807,28 @@ async function finalizeMediaImport(
     jsonMeta.transcodingJobKey = jobBase.key;
   }
 
-  let foundDetections = false;
-
-  /* Look for JSON track file as first priority */
-  const trackFileAbsPath = await _findJsonTrackFile(jsonMeta.originalBasePath);
-  if (trackFileAbsPath !== null) {
-    /* Move the track file into the new project directory */
-    await fs.copy(
-      trackFileAbsPath,
-      npath.join(projectDirAbsPath, npath.basename(trackFileAbsPath)),
-    );
-    //Load tracks to generate attributes
-    const tracks = await loadJsonTracks(trackFileAbsPath);
-    const { attributes } = processTrackAttributes(Object.values(tracks));
-    if (attributes) jsonMeta.attributes = attributes;
-    foundDetections = true;
-  }
-  /* Look for other types of annotation files as a second priority */
-  if (!foundDetections) {
-    const contents = await fs.readdir(jsonMeta.originalBasePath);
-    const csvFileCandidates = contents
-      .filter((v) => CsvFileName.test(v))
-      .map((filename) => npath.join(jsonMeta.originalBasePath, filename));
-    if (csvFileCandidates.length > 1) {
-      throw new Error(`too many CSV files found in ${jsonMeta.originalBasePath}, expected at most 1`);
+  //We need to create datasets for each of the multiCam folders as well
+  if (datasetType === 'multi' && jsonMeta.multiCam?.cameras) {
+    const pairs = Object.entries(jsonMeta.multiCam.cameras);
+    for (let i = 0; i < pairs.length; i += 1) {
+      const val = pairs[i][1];
+      const key = pairs[i][0];
+
+      const jsonClone = { ...cloneDeep(jsonMeta), ...val };
+      jsonClone.multiCam = null;
+      jsonClone.id = `${jsonMeta.id}/${key}`;
+      jsonClone.transcodedVideoFile = val.transcodedVideoFile || '';
+      jsonClone.transcodedImageFiles = val.transcodedImageFiles || [];
+      jsonClone.subType = null;
+
+      // eslint-disable-next-line no-await-in-loop
+      const cameraDirAbsPath = await _initializeProjectDir(settings, jsonClone);
+      // eslint-disable-next-line no-await-in-loop
+      await _importTrackFile(settings, jsonClone.id, cameraDirAbsPath, jsonClone);
```

Comment:
> * What is the benefit of modeling each camera like its own dataset?

Initially I didn't do this until I figured out that each of the currently running pipelines returns an independent annotation file for each camera.  It allows me to re-use some loading and display logic because camera behaves like a standard dataset.  It also allows a user to run independent pipelines on each camera.

> * What happens when we need to support some kind of deterministic transforms between cameras, so that one camera depends on the annotations of another?

I have wondered this but haven't received a definitive example of this yet.  I'm hoping that the eventual 'default camera' would fall under this special case and can be used to collate data across pipelines or run pipelines that create data like this.

> * Should we consider not managing the data like this?  For example, I could imagine that jsonMeta gets some new keys to point directly to annotation files instead of discovering them in a folder.

I am assuming this is about jsonMeta discovery stuff and no the overally each camera being it's own dataset?  I would like this change in the future and I have a sort of pre-discovery used for attaching the annotation file select during the finalize-media-import dialog much like web does.  It would be nice to tie the result of that directly in the jsonMeta instead of relying on the local folder discovery service.

> What are the risks to doing this?

File replacement directly in the project folder could break unless it is the same name which I don't think is that big of  a deal.

Reply:
Thanks for the write-up.  I just wanted something we could refer back to when if/when this has to change.

No change necessary, just wanted a record of the decision.  Thanks.